### PR TITLE
Add AlignUpBySize(...) to align by size given

### PR DESCRIPTION
**Pull request by 10000TB**

### CODE REVIEW

File: pkg/boot/align/align.go [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 // license that can be found in the LICENSE file.
-
 package align
 
 import "os"
 
-var pageMask = uint(os.Getpagesize() - 1)
+var pageSize = uint(os.Getpagesize())
+
+func AlignUpBySize(v uint, alignSize uint) uint {
```
Comment (rminnich) [inline pkg/boot/align/align.go:10]: align.go:1:1: package comment should be of the form "Package align ..."
align.go:10:1: exported function AlignUpBySize should have comment or be unexported
align.go:10:6: func name will be used as align.AlignUpBySize by other packages, and that stutters; consider calling this UpBySize
align.go:19:1: exported function AlignUpPageSize should have comment or be unexported
align.go:19:6: func name will be used as align.AlignUpPageSize by other packages, and that stutters; consider calling this UpPageSize
align.go:23:1: exported function AlignUpPageSizePtr should have comment or be unexported
align.go:23:6: func name will be used as align.AlignUpPageSizePtr by other packages, and that stutters; consider calling this UpPageSizePtr

Comment (rminnich) [inline pkg/boot/align/align.go:10]: Why does this align up to a page size if they ask for less than a page size? What if people need a certain alignment?

You have a nice function (below) that aligns to a page size; why not just let people use that for page size alignment? 

This is a generic function, not related to boot; might it go in the general purpose package in uroot utils?

File: pkg/boot/align/align_test.go [hunk 2]
```diff
@@ -1,42 +1,62 @@
...
+		want      uint
 	}{
 		{
-			name: "below",
-			val:  uint(0x02),
-			want: uint(0x04),
+			name:      "below",
+			val:       uint(0x02),
+			alignSize: uint(0x04),
+			want:      uint(0x04),
```
Comment (rminnich) [inline pkg/boot/align/align_test.go:21]: This test should probably check that the returned alignment is correct.

**Comment by rminnich** [comment]
Generally people push pull requests to their fork, not directly; I am wondering why you pushed to u-root this time?

**Review by rminnich — COMMENTED** [review]
suggestion: 
align.Up
align.Down
align.UpPage
align.DownPage

short and easy to remember

**Comment by rminnich** [comment]
btw, why is align.Up ok in my view? 
because go doc align.Up will make it clear, by the parameters, what it does in more detail.

**Review by rminnich — COMMENTED** [review]
I'm going to remove this branch in a bit, please follow the process and send PRs from your fork. Thanks

**Comment by rminnich** [comment]
please generate a PR from a fork of this repo, and please respond to golint and other comments. Also, consider whether it needs to be in boot/ or u-root/ package.

File: pkg/boot/align/align.go (hunk 1 above)
Comment (10000TB) [reply pkg/boot/align/align.go:10]: Done

File: pkg/boot/align/align_test.go (hunk 2 above)
Comment (10000TB) [reply pkg/boot/align/align_test.go:21]: Using `want` as the expected return. Does that make sense to you ?

**Pull request closed without merging**